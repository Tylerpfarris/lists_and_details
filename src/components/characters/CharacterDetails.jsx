import React, { Component } from 'react';
import { findById } from '../../services/findById';
import { getRandomGIF } from '../../services/getRandomGIF';
export default class CharacterDetails extends Component {
  state = {
    character: {},
    loading: true,
    gif: [],
  };

  async componentDidMount() {
    try {
      const id = await this.props.match.params.id;
      const character = await findById(id);
      const gif = await getRandomGIF();
      this.setState({
        character,
        loading: false,
        gif,
      });
    } catch (error) {
      throw Error(error);
    }
  }

  render() {
    const { image, name } = this.state.character;
    const { loading, gif } = this.state;
    return loading ? (
      <h1>Loading...</h1>
    ) : (
      <>
        <div role="display" aria-label="character details">
          <h1>{name}</h1>
          <img src={image} alt={name} />
        </div>
        <h3>A Random GIF</h3>
        <img src={gif.gifLink} alt={gif._id} />
      </>
    );
  }
}
