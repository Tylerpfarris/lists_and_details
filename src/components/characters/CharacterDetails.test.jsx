import React from 'react';
import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import CharacterDetails from './CharacterDetails';

describe('CharacterDetails', () => {
  it('renders a character from the id onto the page ', async () => {
    render(
      <MemoryRouter>
        <CharacterDetails
          match={{
            params: {
              id: '5da237699734fdcb7bef8f7c',
            },
          }}
        />
      </MemoryRouter>
    );
      
    screen.getByText('Loading...');

    const divEl = await screen.findByRole('display', {
      name: 'character details',
    });
    expect(divEl).toMatchSnapshot();
  });
});
