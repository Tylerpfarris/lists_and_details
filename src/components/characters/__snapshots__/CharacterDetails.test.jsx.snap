// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharacterDetails renders a character from the id onto the page  1`] = `
<div
  aria-label="character details"
  role="display"
>
  <h1>
    Uncle Stinky
  </h1>
  <img
    alt="Uncle Stinky"
    src="https://vignette.wikia.nocookie.net/heyarnold/images/8/89/Uncle_Stinky.jpg/revision/latest/scale-to-width-down/310?cb=20120924204406"
  />
</div>
`;
